Basic use case examples not working: `fixture 'message' not found`

**Describe the bug**
Following the basic use cases in the README does not work.
pytest returns `fixture 'message' not found`.
Tested with poetry and pip.

**To Reproduce**
```bash
poetry init
poetry add mockafka-py
poetry shell
```

```python
### test.py ###
from mockafka import produce, consume

@produce(topic='test', key='test_key', value='test_value', partition=4)
@consume(topics=['test'])
def test_produce_and_consume_decorator(message):
    """
    This test showcases the usage of both @produce and @consume decorators in a single test case.
    It produces a message to the 'test' topic and then consumes it to perform further logic.
    # Notice you may get message None
    """
    # Your test logic for processing the consumed message here

    if not message:
        return

    pass
```

```bash
$ pytest test.py

=================================================================================================================================================== test session starts ====================================================================================================================================================
platform linux -- Python 3.11.6, pytest-8.3.3, pluggy-1.5.0
rootdir: 
configfile: pyproject.toml
plugins: asyncio-0.23.8, cov-5.0.0
asyncio: mode=Mode.STRICT
collected 1 item

test.py E                                                                                                                                                                                                                                                                                                            [100%]

========================================================================================================================================================== ERRORS ==========================================================================================================================================================
___________________________________________________________________________________________________________________________________ ERROR at setup of test_produce_and_consume_decorator ___________________________________________________________________________________________________________________________________
file test.py, line 4
  @produce(topic='test', key='test_key', value='test_value', partition=4)
  @consume(topics=['test'])
  def test_produce_and_consume_decorator(message):
E       fixture 'message' not found
>       available fixtures: _session_event_loop, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, doctest_namespace, event_loop, event_loop_policy, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, test.py::<event_loop>, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, unused_tcp_port, unused_tcp_port_factory, unused_udp_port, unused_udp_port_factory
>       use 'pytest --fixtures [testpath]' for help on them.
```

**Expected behavior**
Test passes successfully.

**Desktop (please complete the following information):**
 - OS: Ubuntu 23.10
 - python version 3.11.6
 - mockafka-py:  0.1.61

--- a/.gitignore
+++ b/.gitignore
@@ -10,6 +10,7 @@
 .idea/**/shelf
 .idea/*
 .vscode
+.DS_Store
 
 # AWS User-specific
 .idea/**/aws.xml

--- a/mockafka/decorators/consumer.py
+++ b/mockafka/decorators/consumer.py
@@ -1,5 +1,6 @@
 from __future__ import annotations
 
+import inspect
 from functools import wraps
 from mockafka import FakeConsumer
 
@@ -21,6 +22,21 @@ def consume(topics: list[str], auto_commit: bool = True):
     """
 
     def decorator(func):
+        # Get the function signature
+        sig = inspect.signature(func)
+        
+        # Create a new signature with default values for message parameters
+        new_params = []
+        for param_name, param in sig.parameters.items():
+            if param_name == 'message' and param.default == inspect.Parameter.empty:
+                # Add default value to prevent pytest from treating it as a fixture
+                new_param = param.replace(default=None)
+                new_params.append(new_param)
+            else:
+                new_params.append(param)
+        
+        new_sig = sig.replace(parameters=new_params)
+        
         @wraps(func)
         def wrapper(*args, **kwargs):
             # Create a FakeConsumer instance and subscribe to specified topics
@@ -39,9 +55,11 @@ def consume(topics: list[str], auto_commit: bool = True):
                 func(message=message, *args, **kwargs)
 
             # Call the original function without a message parameter
-            result = func(*args, **kwargs)
+            result = func(message=None, *args, **kwargs)
             return result
 
+        # Apply the new signature to the wrapper
+        wrapper.__signature__ = new_sig
         return wrapper
 
     return decorator

--- a/tests/test_docrators.py
+++ b/tests/test_docrators.py
@@ -113,3 +113,119 @@ class TestDecorators(TestCase):
 
         self.assertEqual(message.key(), "test_")
         self.assertEqual(message._partition, 5)
+
+    # NEW TEST CASES FOR PYTEST FIXTURE ISSUE FIX
+
+    @setup_kafka(topics=[{"topic": "pytest_fix_test", "partition": 16}])
+    @produce(topic="pytest_fix_test", key="fix_key", value="fix_value", partition=3)
+    @consume(topics=["pytest_fix_test"])
+    def test_consume_decorator_without_default_parameter(self, message):
+        """
+        Test case for the original pytest fixture issue.
+        This test verifies that @consume decorator works with function parameters
+        that don't have default values, which used to cause pytest to treat 'message'
+        as a fixture and fail with 'fixture message not found'.
+        
+        This reproduces the exact issue from the bug report.
+        """
+        if message is None:
+            return
+
+        # Verify the message was consumed correctly
+        self.assertEqual(message.key(), "fix_key")
+        self.assertEqual(message.value(payload=None), "fix_value")
+        self.assertEqual(message.partition(), 3)
+
+    @setup_kafka(topics=[{"topic": "pytest_fix_test2", "partition": 16}])
+    @produce(topic="pytest_fix_test2", key="fix_key2", value="fix_value2", partition=7)
+    @consume(topics=["pytest_fix_test2"])
+    def test_consume_decorator_with_explicit_default(self, message=None):
+        """
+        Test case to ensure backward compatibility.
+        This verifies that functions with explicit default values still work correctly
+        after the pytest fixture fix.
+        """
+        if message is None:
+            return
+
+        # Verify the message was consumed correctly
+        self.assertEqual(message.key(), "fix_key2")
+        self.assertEqual(message.value(payload=None), "fix_value2")
+        self.assertEqual(message.partition(), 7)
+
+    @setup_kafka(topics=[{"topic": "multiple_messages_test", "partition": 16}])
+    @produce(topic="multiple_messages_test", key="msg1_key", value="msg1_value", partition=1)
+    @produce(topic="multiple_messages_test", key="msg2_key", value="msg2_value", partition=2)
+    @produce(topic="multiple_messages_test", key="msg3_key", value="msg3_value", partition=3)
+    @consume(topics=["multiple_messages_test"])
+    def test_consume_decorator_multiple_messages_no_default(self, message):
+        """
+        Test case for multiple message consumption without default parameter.
+        This ensures the fix works correctly when multiple messages are produced
+        and consumed, and the function parameter has no default value.
+        """
+        if message is None:
+            return
+
+        # Verify we receive one of the expected messages
+        expected_keys = ["msg1_key", "msg2_key", "msg3_key"]
+        expected_values = ["msg1_value", "msg2_value", "msg3_value"]
+        expected_partitions = [1, 2, 3]
+
+        self.assertIn(message.key(), expected_keys)
+        self.assertIn(message.value(payload=None), expected_values)
+        self.assertIn(message.partition(), expected_partitions)
+
+    @setup_kafka(topics=[{"topic": "edge_case_test", "partition": 16}])
+    @produce(topic="edge_case_test", key="edge_key", value="edge_value", partition=0)
+    @consume(topics=["edge_case_test"])
+    def test_consume_decorator_edge_case_partition_zero(self, message):
+        """
+        Test case for edge case with partition 0.
+        This ensures the fix works correctly even with partition 0,
+        which can sometimes be a special case in Kafka implementations.
+        """
+        if message is None:
+            return
+
+        self.assertEqual(message.key(), "edge_key")
+        self.assertEqual(message.value(payload=None), "edge_value")
+        self.assertEqual(message.partition(), 0)
+
+    @setup_kafka(topics=[{"topic": "signature_test", "partition": 16}])
+    @produce(topic="signature_test", key="sig_key", value="sig_value", partition=5)
+    @consume(topics=["signature_test"])
+    def test_consume_decorator_function_signature_modification(self, message):
+        """
+        Test case to verify that the function works correctly with the signature modification.
+        This test ensures that the decorator correctly handles the message parameter
+        without causing pytest fixture issues.
+        """
+        if message is None:
+            return
+
+        self.assertEqual(message.key(), "sig_key")
+        self.assertEqual(message.value(payload=None), "sig_value")
+        self.assertEqual(message.partition(), 5)
+
+    def test_signature_modification_verification(self):
+        """
+        Separate test to verify that the decorator modifies function signatures correctly.
+        This test creates a decorated function and inspects its signature.
+        """
+        import inspect
+        
+        @consume(topics=["test_topic"])
+        def test_function_for_signature(message):
+            pass
+        
+        # Get the signature of the decorated function
+        sig = inspect.signature(test_function_for_signature)
+        
+        # Verify that the message parameter exists
+        message_param = sig.parameters.get('message')
+        self.assertIsNotNone(message_param)
+        
+        # The test should pass if the default is not empty (even if it's None)
+        # This prevents pytest from treating 'message' as a fixture
+        self.assertNotEqual(message_param.default, inspect.Parameter.empty)

--- a/tests/test_pytest_fixture_issue.py
+++ b/tests/test_pytest_fixture_issue.py
@@ -1,0 +1,217 @@
+"""
+Test cases specifically for the pytest fixture issue fix in @consume decorator.
+
+This module contains tests that reproduce and verify the fix for the issue where
+pytest would treat function parameters without default values as fixtures,
+causing 'fixture not found' errors when using the @consume decorator.
+
+Issue: https://github.com/aiven/mockafka-py/issues/175
+"""
+
+from __future__ import annotations
+
+import inspect
+from unittest import TestCase
+
+from mockafka import produce, consume, setup_kafka, Message
+
+
+class TestPytestFixtureIssueFix(TestCase):
+    """Test cases for the pytest fixture issue fix in @consume decorator."""
+
+    def test_original_bug_report_example(self):
+        """
+        Test the exact example from the bug report.
+        
+        This test reproduces the original issue where the following code
+        would fail with 'fixture message not found':
+        
+        @produce(topic='test', key='test_key', value='test_value', partition=4)
+        @consume(topics=['test'])
+        def test_function(message):  # No default value
+            pass
+        """
+        
+        @setup_kafka(topics=[{"topic": "bug_report_test", "partition": 16}])
+        @produce(topic='bug_report_test', key='test_key', value='test_value', partition=4)
+        @consume(topics=['bug_report_test'])
+        def test_produce_and_consume_decorator(message):
+            """
+            This test showcases the usage of both @produce and @consume decorators in a single test case.
+            It produces a message to the 'test' topic and then consumes it to perform further logic.
+            # Notice you may get message None
+            """
+            if not message:
+                return
+
+            # Verify the message content
+            self.assertEqual(message.key(), 'test_key')
+            self.assertEqual(message.value(payload=None), 'test_value')
+            self.assertEqual(message.partition(), 4)
+
+        # This should not raise any pytest fixture errors
+        test_produce_and_consume_decorator()
+
+    def test_function_signature_modification(self):
+        """
+        Test that the @consume decorator properly modifies function signatures.
+        
+        The decorator should add default values to 'message' parameters that don't
+        have them, preventing pytest from treating them as fixtures.
+        """
+        
+        @consume(topics=['test_topic'])
+        def test_function_no_default(message):
+            pass
+        
+        @consume(topics=['test_topic'])
+        def test_function_with_default(message=None):
+            pass
+        
+        # Check that both functions now have default values for message parameter
+        sig1 = inspect.signature(test_function_no_default)
+        sig2 = inspect.signature(test_function_with_default)
+        
+        message_param1 = sig1.parameters.get('message')
+        message_param2 = sig2.parameters.get('message')
+        
+        self.assertIsNotNone(message_param1)
+        self.assertIsNotNone(message_param2)
+        
+        # Both should have default values now (not empty, even if None)
+        self.assertNotEqual(message_param1.default, inspect.Parameter.empty)
+        self.assertNotEqual(message_param2.default, inspect.Parameter.empty)
+
+    def test_multiple_parameters_with_message(self):
+        """
+        Test that the decorator works correctly with functions that have
+        multiple parameters, where only 'message' needs modification.
+        """
+        
+        @consume(topics=['test_topic'])
+        def test_function_multiple_params(self, message, other_param="default"):
+            pass
+        
+        sig = inspect.signature(test_function_multiple_params)
+        
+        # Check that message parameter has default value
+        message_param = sig.parameters.get('message')
+        self.assertIsNotNone(message_param)
+        self.assertNotEqual(message_param.default, inspect.Parameter.empty)
+        
+        # Check that other parameters are unchanged
+        self_param = sig.parameters.get('self')
+        other_param = sig.parameters.get('other_param')
+        
+        self.assertIsNotNone(self_param)
+        self.assertIsNotNone(other_param)
+        self.assertEqual(other_param.default, "default")
+
+    def test_no_message_parameter(self):
+        """
+        Test that the decorator works correctly with functions that don't
+        have a 'message' parameter at all.
+        """
+        
+        @consume(topics=['test_topic'])
+        def test_function_no_message_param(self, other_param):
+            pass
+        
+        sig = inspect.signature(test_function_no_message_param)
+        
+        # Should not have a message parameter
+        message_param = sig.parameters.get('message')
+        self.assertIsNone(message_param)
+        
+        # Other parameters should be unchanged
+        self_param = sig.parameters.get('self')
+        other_param = sig.parameters.get('other_param')
+        
+        self.assertIsNotNone(self_param)
+        self.assertIsNotNone(other_param)
+        # other_param should not have a default value
+        self.assertEqual(other_param.default, inspect.Parameter.empty)
+
+    def test_backward_compatibility_with_existing_tests(self):
+        """
+        Test that the fix maintains backward compatibility with existing test patterns.
+        
+        This ensures that existing tests that already have default values continue to work.
+        """
+        
+        @setup_kafka(topics=[{"topic": "compat_test", "partition": 16}])
+        @produce(topic="compat_test", key="compat_key", value="compat_value", partition=1)
+        @consume(topics=["compat_test"])
+        def test_existing_pattern(message: Message | None = None):
+            if message is None:
+                return
+            
+            self.assertEqual(message.key(), "compat_key")
+            self.assertEqual(message.value(payload=None), "compat_value")
+            self.assertEqual(message.partition(), 1)
+        
+        # This should work without any issues
+        test_existing_pattern()
+
+    def test_edge_case_empty_function_signature(self):
+        """
+        Test edge case where function has no parameters at all.
+        """
+        
+        @consume(topics=['test_topic'])
+        def test_function_no_params():
+            pass
+        
+        sig = inspect.signature(test_function_no_params)
+        
+        # Should have no parameters
+        self.assertEqual(len(sig.parameters), 0)
+
+    def test_complex_scenario_multiple_decorators(self):
+        """
+        Test complex scenario with multiple decorators and message consumption.
+        
+        This test ensures the fix works in realistic scenarios where multiple
+        decorators are stacked together.
+        """
+        
+        @setup_kafka(topics=[{"topic": "complex_test", "partition": 16}])
+        @produce(topic="complex_test", key="key1", value="value1", partition=1)
+        @produce(topic="complex_test", key="key2", value="value2", partition=2)
+        @consume(topics=["complex_test"])
+        def test_complex_scenario(message):
+            if message is None:
+                return
+            
+            # Should receive one of the two messages
+            expected_data = [
+                ("key1", "value1", 1),
+                ("key2", "value2", 2)
+            ]
+            
+            actual_data = (message.key(), message.value(payload=None), message.partition())
+            self.assertIn(actual_data, expected_data)
+        
+        # This should work without pytest fixture errors
+        test_complex_scenario()
+
+    def test_function_with_args_and_kwargs(self):
+        """
+        Test that the decorator works with functions that use *args and **kwargs.
+        """
+        
+        @consume(topics=['test_topic'])
+        def test_function_with_varargs(message, *args, **kwargs):
+            pass
+        
+        sig = inspect.signature(test_function_with_varargs)
+        
+        # Check that message parameter has default value
+        message_param = sig.parameters.get('message')
+        self.assertIsNotNone(message_param)
+        self.assertNotEqual(message_param.default, inspect.Parameter.empty)
+        
+        # Check that *args and **kwargs are preserved
+        params = list(sig.parameters.keys())
+        self.assertIn('args', params)
+        self.assertIn('kwargs', params) 

